text/0000-guaranteed-slice-repr.md

```diff
@@ -0,0 +1,182 @@
+- Feature Name: guaranteed_slice_repr
+- Start Date: 2025-02-18
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC guarantees the in-memory representation of slice and str references.
+Specifically, `&[T]` and `&mut [T]` are guaranteed to have the same layout as:
```

Comment:
do these apply to `*const [T]` and `*mut [T]`?

Reply:
We have stable docs that "pointers and references have the same layout" ([src](https://doc.rust-lang.org/reference/type-layout.html#r-layout.pointer)), so I think it must.